WalletWasabi.Fluent/ViewModels/Wallets/Send/ChangeAvoidanceSuggestionViewModel.cs

```diff
@@ -100,6 +102,30 @@ public partial class ChangeAvoidanceSuggestionViewModel : SuggestionViewModel
 				wallet.Synchronizer.UsdExchangeRate, false);
 		}
 
+		ChangeAvoidanceSuggestionViewModel? bnbSuggestion = null;
+
+		BranchAndBound bnb = new(wallet.Coins.Confirmed().ToList());
+
+		bool canBuildWithoutChangeWithBnb = bnb.TryGetExactMatch(transactionInfo.Amount, out List<SmartCoin>? bnbResult);
+
+		if (canBuildWithoutChangeWithBnb && bnbResult is not null)
+		{
+			var bnbTransaction = await Task.Run(() => TransactionHelpers.BuildChangelessTransaction(
+				wallet,
+				transactionInfo.Address,
+				transactionInfo.UserLabels,
+				transactionInfo.FeeRate,
+				bnbResult
+				));
```

Comment:
We build up a transaction here and after sending it and checking it on `blockstream.info/testnet/`, the TX looks nothing like the original.

Need to figure out something to solve this.

Reply:
> We build up a transaction here and after sending it and checking it on blockstream.info/testnet/, the TX looks nothing like the original.

What does this mean?

Reply:
In the codebase we have a feature that makes sure if we use one UTXO out of a cluster/pocket/idk (I'm really not sure about the details), the "unnecessary" UTXOs will be merged together and sent to our generated change address.

I think this is what happens here as well:

- We build up a changless transaction.
- When we finalize the Tx, we detect that not every UTXO of the cluster/pocket is used in the transaction.
- Wasabi filles in the remaining UTXOs and generate a change address for them.
- Now our transaction is not changless and a developer (@adamPetho) gets confused.

Reply:
This should be fixed now.

Reply:
What you are saying sounds like you hit this branch: 

https://github.com/zkSNACKs/WalletWasabi/blob/ff33bcacba7c0021fbf72b27dcd462668cde8b34/WalletWasabi/Blockchain/Transactions/TransactionFactory.cs#L86

and even though 

https://github.com/zkSNACKs/WalletWasabi/blob/ff33bcacba7c0021fbf72b27dcd462668cde8b34/WalletWasabi.Fluent/Helpers/TransactionHelpers.cs#L24

suggests that `ChangeStrategy` is `ChangeStrategy.AllRemainingCustom`, it might not be the case because of

https://github.com/zkSNACKs/WalletWasabi/blob/ff33bcacba7c0021fbf72b27dcd462668cde8b34/WalletWasabi/Blockchain/TransactionBuilding/PaymentIntent.cs#L61

It's just a hunch. Please tell me if it is possible or not. IDK but it sounds like your issue really.